src/sentry/scim/endpoints/members.py

```diff
@@ -250,6 +251,83 @@ def delete(self, request: Request, organization, member) -> Response:
         self._delete_member(request, organization, member)
         return Response(status=204)
 
+    @extend_schema(
+        operation_id="Update an Organization Member's Attributes",
+        parameters=[GLOBAL_PARAMS.ORG_SLUG, SCIM_PARAMS.MEMBER_ID],
+        request=inline_serializer(
+            "SCIMMemberProvision", fields={"sentryOrgRole": serializers.CharField()}
+        ),
+        responses={
+            201: OrganizationMemberSCIMSerializer,
+            401: RESPONSE_UNAUTHORIZED,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOTFOUND,
+        },
+        examples=[  # TODO: see if this can go on serializer object instead
+            OpenApiExample(
+                "Update a user",
+                response_only=True,
+                value={
+                    "schemas": ["urn:ietf:params:scim:schemas:core:2.0:User"],
+                    "id": "242",
+                    "userName": "test.user@okta.local",
+                    "emails": [{"primary": True, "value": "test.user@okta.local", "type": "work"}],
+                    "active": True,
+                    "name": {"familyName": "N/A", "givenName": "N/A"},
+                    "meta": {"resourceType": "User"},
+                },
+                status_codes=["201"],
+            ),
+        ],
+    )
+    def put(self, request: Request, organization, member):
+        """
+        Update an organization member
+
+        Currently only updates organization role
+        """
+        if features.has("organizations:scim-orgmember-roles", organization, actor=None):
+
+            if "sentryOrgRole" in request.data and request.data["sentryOrgRole"]:
+                # Only do an update if something is changing
+                if (
+                    member.flags["idp:role-restricted"]
+                    and member.role.lower() == request.data["sentryOrgRole"].lower()
+                ):
+                    context = serialize(
+                        member, serializer=_scim_member_serializer_with_expansion(organization)
+                    )
+                    return Response(context, status=200)
+                requested_role = request.data["sentryOrgRole"].lower()
+                idp_role_restricted = True
+            else:
+                # Only reset role if current role isn't the default
```

Comment:
will we allow for the case where a member is currently idp:role-restricted and we want to edit them to have a blank role (and thereby remove the idp:role-restricted flag?)

Reply:
Yeah, my thinking is that if the user explicitly sets the role to blank, then they don't want this user's role to be fixed by the IDP